src/generic_datatype.cpp

```diff
@@ -0,0 +1,101 @@
+#include "generic_datatype.h"
+
+#include <memory>
+#include <unordered_map>
+
+#include "assert.h"
+
+using namespace std;
+
+namespace smt {
+GenericDatatypeConstructorDecl::GenericDatatypeConstructorDecl(
+    const std::string & name)
+    : cons_name(name)
+{
+}
+
+void GenericDatatypeConstructorDecl::add_new_selector(
+    const shared_ptr<selectorComponents> & newSelector)
```

Comment:
```suggestion
    const selectorComponents & newSelector)
```
(and required changes).